src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php

```diff
@@ -424,4 +519,30 @@ private function hasProductImage(Response $response, ProductInterface $product):
 
         return false;
     }
+
+    private function getFieldValueOfFirstProduct(array $product, string $field): ?string
+    {
+        if ($field === 'code') {
+            return $product['code'];
+        }
+
+        if ($field === 'name') {
+            return $product['translations']['en_US']['name'];
+        }
+
+        return null;
+    }
+
+    private function hasProductWithFieldValue(Response $response, string $field, string $value): bool
```

Comment:
IMO having two definitions instead of unified `product with (name|code) "([^"]+)" should not be added` is less smelly than having this private function in the context